xbmc/threads/Thread.cpp

```diff
@@ -102,6 +102,7 @@ void CThread::Create(bool bAutoDelete, unsigned stacksize)
 
 bool CThread::IsRunning() const
 {
+  CSingleLock lock(m_CriticalSection);
```

Comment:
@neo1973 is correct; the lock/atomic is required here. Whether or not you are interested in the id have nothing to do with it. Without synchronization the value may not even be visible to readers.